packages/react-components/src/trace-explorer/trace-explorer-time-range-data-widget.tsx

```diff
@@ -267,6 +267,19 @@ export class ReactTimeRangeDataWidget extends React.Component<
         }
     };
 
+    /**
+     * Allows public to set active data and experiment data map.
+     * This is used in vscode-trace-extension where we need to manually
+     * persist and restore data using the VSCode API when component is
+     * not visible.  User inputs are not restored.
+     * @param activeData
+     * @param map
+     */
+    public restoreState = (activeData: ExperimentTimeRangeData, map: Map<string, ExperimentTimeRangeData>) => {
```

Comment:
can't you just send a signal to itself when it's time to restore as if the signal came from the viewer panel?

Reply:
Yes, this can be accessed by a signal.